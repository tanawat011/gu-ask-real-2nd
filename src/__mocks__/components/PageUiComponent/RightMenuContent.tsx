// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck

export const RightMenuContent = () => {
  return <div data-testid='right-menu-content'>RightMenuContent</div>
}
