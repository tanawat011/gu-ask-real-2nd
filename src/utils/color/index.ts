export * from './hex2Rgb'
export * from './rgb2Hex'
