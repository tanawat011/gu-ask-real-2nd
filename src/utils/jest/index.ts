export * from './renderWithProviders'
export * from './twColor'
