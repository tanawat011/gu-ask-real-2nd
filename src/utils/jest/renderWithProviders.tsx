import type { RenderResult } from '@testing-library/react'

import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { RecoilRoot } from 'recoil'

const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  const queryClient = new QueryClient()

  return (
    <QueryClientProvider client={queryClient}>
      <RecoilRoot>
        <BrowserRouter>{children}</BrowserRouter>
      </RecoilRoot>
    </QueryClientProvider>
  )
}

interface RenderWithProvidersOptions {
  router?: {
    route: string
  }
  recoil?: object
}

export const renderWithProviders = (
  ui: JSX.Element,
  option?: RenderWithProvidersOptions,
): RenderResult => {
  const routerOption = { ...(option?.router || { route: '/' }) }
  window.history.pushState({}, 'Test page', routerOption.route)

  return {
    ...render(ui, { wrapper: AllTheProviders, ...routerOption }),
  }
}
