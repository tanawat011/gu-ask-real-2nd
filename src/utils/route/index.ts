export * from './menuGenerator'
export * from './renderRoutes'
export * from './routeGenerator'
