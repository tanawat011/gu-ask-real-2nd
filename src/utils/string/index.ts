export * from './camelToKebab'
export * from './camelToSentence'
export * from './replaceAll'
