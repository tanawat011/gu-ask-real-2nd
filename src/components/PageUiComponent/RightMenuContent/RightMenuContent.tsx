import React from 'react'

import { Link } from 'react-scroll'
import tw, { css, styled } from 'twin.macro'

import { Title } from 'components/PageUiComponent'
import { TEXT_THEME } from 'constants/twTheme'

type RightMenuContentProps = {
  title: string
  menuList: {
    title: string
  }[]
}

const TwUl = styled.ul(() => [TEXT_THEME.DETAIL, tw`grid gap-2`])
const TwLink = styled(Link)(() => [
  tw`cursor-pointer select-none hover:(font-semibold text-gray-900) dark:hover:text-gray-50`,
  css`
    &.active {
      ${tw`font-semibold text-gray-900 dark:text-gray-50`}
    }
  `,
])

export const RightMenuContent: React.FC<RightMenuContentProps> = ({
  title,
  menuList,
}) => {
  return (
    <>
      <Title size='xs' title={title} />

      <TwUl>
        {menuList.map((item, index) => {
          const { title: text } = item

          const id = text.replaceAll(' ', '-').toLowerCase()

          return (
            <li key={`${index}`}>
              <TwLink
                activeClass='active'
                to={id}
                spy={true}
                isDynamic
                containerId='main-layout'
                offset={-16}
              >
                {text}
              </TwLink>
            </li>
          )
        })}
      </TwUl>
    </>
  )
}
