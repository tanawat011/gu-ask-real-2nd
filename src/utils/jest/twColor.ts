import { hex2Rgb } from 'utils/color'

export const twColor = (hexColor: string, type?: 'bg' | 'border' | 'text') => {
  if (hexColor === 'transparent') {
    return 'transparent'
  }

  if (hexColor.length > 7) {
    return 'undefined'
  }

  const rgb = hex2Rgb(hexColor, { isArray: true }) as string[]

  if (!type) {
    type = 'bg'
  }

  return `rgb(${rgb.join(' ')} / var(--tw-${type}-opacity))`
}
