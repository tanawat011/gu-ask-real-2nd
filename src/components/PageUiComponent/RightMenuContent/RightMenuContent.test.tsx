import { renderWithProviders } from 'utils/jest'

import { RightMenuContent } from './RightMenuContent'

jest.mock('../Title', () => ({
  Title: () => <div>Title</div>,
}))

describe('<RightMenuContent />', () => {
  test('renders correctly', () => {
    expect.assertions(1)

    const { asFragment } = renderWithProviders(
      <RightMenuContent title='Title' menuList={['1', '2', '3']} />,
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
