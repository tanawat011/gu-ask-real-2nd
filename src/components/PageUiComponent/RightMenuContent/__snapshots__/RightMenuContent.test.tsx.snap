// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RightMenuContent /> renders correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="title"
  >
    Title
  </div>
  <ul
    class="grid gap-2"
  >
    <li>
      <a
        href="#"
      >
        1
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        2
      </a>
    </li>
    <li>
      <a
        href="#"
      >
        3
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
